store/sqlstore/channel_store.go

```diff
@@ -416,6 +445,188 @@ func (s SqlChannelStore) createIndexesIfNotExists() {
+			DisplayName: "Favorites",
+			Id:          model.NewId(),
+			UserId:      u.UserId,
+			TeamId:      u.TeamId,
+			Order:       0,
+			Type:        SidebarCategoryFavorites,
+		}); err != nil && !IsUniqueConstraintError(err, []string{"UserId"}) {
+			return model.NewAppError("SqlChannelStore.MigrateFavoriteChannels", "store.sql_channel.save.open_transaction.app_error", nil, err.Error(), http.StatusInternalServerError)
+		}
+		if err := transaction.Insert(&sidebarCategory{
+			DisplayName: "DM",
+			Id:          model.NewId(),
+			UserId:      u.UserId,
+			TeamId:      u.TeamId,
+			Order:       1,
+			Type:        SidebarCategoryDirectMessages,
+		}); err != nil && !IsUniqueConstraintError(err, []string{"UserId"}) {
+			return model.NewAppError("SqlChannelStore.MigrateFavoriteChannels", "store.sql_channel.save.open_transaction.app_error", nil, err.Error(), http.StatusInternalServerError)
+		}
+
+		if err := transaction.Insert(&sidebarCategory{
+			DisplayName: "Channels",
+			Id:          model.NewId(),
+			UserId:      u.UserId,
+			TeamId:      u.TeamId,
+			Order:       2,
+			Type:        SidebarCategoryChannels,
+		}); err != nil && !IsUniqueConstraintError(err, []string{"UserId"}) {
+			return model.NewAppError("SqlChannelStore.MigrateFavoriteChannels", "store.sql_channel.save.open_transaction.app_error", nil, err.Error(), http.StatusInternalServerError)
+		}
+	}
+	if err := transaction.Commit(); err != nil {
+		return model.NewAppError("SqlChannelStore.Save", "store.sql_channel.save.commit_transaction.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+	return nil
+}
+
+// migrateFavoriteChannels copies favorites from Preferences table to SidebarChannels
+func (s SqlChannelStore) migrateFavoriteChannelToSidebarChannels() error {
+	sb := s.
+		getQueryBuilder().
+		Select("Preferences.UserId", "Preferences.Name AS ChannelId", "SidebarCategories.Id AS CategoryId", "0 AS `Order`").
+		From("Preferences").
+		Where(sq.And{
+			sq.Eq{"Preferences.Category": "favorite_channel"},
+			sq.NotEq{"SidebarCategories.Id": nil},
+		}).
+		LeftJoin("Channels ON (Channels.Id=Preferences.Name)").
+		LeftJoin("SidebarCategories ON (SidebarCategories.UserId=Preferences.UserId AND SidebarCategories.TeamId=Channels.TeamId AND SidebarCategories.Type='F')")
+
+	ib := s.getQueryBuilder().
+		Insert("SidebarChannels").
+		Columns("ChannelId", "UserId", "CategoryId", "`Order`").
+		Select(sb)
+
+	if s.DriverName() == model.DATABASE_DRIVER_MYSQL {
+		ib = ib.Suffix("ON DUPLICATE KEY UPDATE UserId=SidebarChannels.UserId")
+	} else {
+		ib = ib.Suffix("ON CONFLICT (UserId) DO NOTHING")
+	}
+	sql, args, err := ib.ToSql()
+	if err != nil {
+		return err
+	}
+	if _, err := s.GetMaster().Exec(sql, args...); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// migrateDirectMessagesToSidebarChannels populates direct messages category with DMs and GMs
+func (s SqlChannelStore) migrateDirectMessagesToSidebarChannels() error {
+	sb := s.
```

Comment:
If we're populating the DM categories, this needs to look at the Preferences table since that's used to hide DMs when the user removes them from their sidebar.

Alternatively, since the DMs channel needs to be sorted dynamically, we could populate it dynamically. The Direct Messages category only offers automatic sorting unlike the other categories that are all manually sorted.